packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/event_client/queued_item_store/dart_queued_item_store.stub.dart

```diff
@@ -0,0 +1,46 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'dart:async';
+
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/event_client/queued_item_store/queued_item_store.dart';
+import 'package:aws_common/aws_common.dart';
+import 'package:drift/drift.dart';
+import 'package:meta/meta.dart';
+
+/// {@template amplify_analytics_pinpoint_dart.dart_queued_item_store}
+/// Stores strings using IndexedDB for web and Drift for all other platforms
+/// {@endtemplate}
+class DartQueuedItemStore implements QueuedItemStore, Closeable {
+  /// {@macro amplify_analytics_pinpoint_dart.dart_queued_item_store}
+  // ignore: avoid_unused_constructor_parameters
+  DartQueuedItemStore(QueryExecutor? driftQueryExecutor) {
```

Comment:
Can we avoid passing in something specific to Drift? I think we can keep all the drift impl details within the the Drift impl of DartQueuedItemStore.

I think there are a few things we can simplify now that we are not use Drift on web.

If you update this constructor to pass in the storage path (which you are already retrieving in the configure fn, where DartQueuedItemStore is instantiated) instead of the QueryExecutor , you can import `connect` from amplify_db_common_dart from within the Drift impl of DartQueuedItemStore to create the QueryExecutor.

I think you can also drop the dependency on amplify_db_common and just depend on amplify_db_common_dart as long as you are retrieving the storage path in configure (which you currently are). You can then remove the connect arg from the `AmplifyAnalyticsPinpointDart` constructor to simplify it a bit.

Reply:
I've implemented your changes.  

A downside of this change is that the connect function is no longer injectable so it cannot be configured or controlled within tests as easily.  We'd need to have a lower level test to change that behavior.